engine/baml-lib/baml-core/src/ir/repr.rs

```diff
@@ -1404,6 +1396,27 @@ pub struct Node<T> {
     pub elem: T,
 }
 
+impl<T> Node<T> {
+    pub fn span(&self) -> Option<&ast::Span> {
+        self.attributes.span.as_ref()
+    }
+
+    pub fn alias(&self, ctx: &EvaluationContext<'_>) -> Result<Option<String>> {
+        self.attributes.alias().map(|v| v.resolve(ctx)).transpose()
+    }
+
+    pub fn description(&self, ctx: &EvaluationContext<'_>) -> Result<Option<String>> {
+        self.attributes
+            .description()
+            .map(|v| v.resolve(ctx))
+            .transpose()
+    }
+
+    pub fn streaming_behavior(&self) -> StreamingBehavior {
+        self.attributes.streaming_behavior()
+    }
+}
+
```

Comment:
These are generic to all node types. Don't have to be copy pasted on every single walker.